bundles/core/org.eclipse.smarthome.core.audio/src/main/java/org/eclipse/smarthome/core/audio/AudioManager.java

```diff
@@ -0,0 +1,460 @@
+                    logger.error("Error playing '{}': {}", audioStream.toString(), e.getMessage());
+                }
+            } else {
+                logger.warn("Failed playing audio stream '{}' as no audio sink was found.", audioStream.toString());
+            }
+        }
+    }
+
+    /**
+     * Plays an audio file from the "sounds" folder using the default audio sink.
+     *
+     * @throws AudioException in case the file does not exist or cannot be opened
+     */
+    public void playFile(String fileName) throws AudioException {
+        playFile(fileName, null);
+    }
+
+    /**
+     * Plays an audio file from the "sounds" folder using the given audio sink.
+     *
+     * @throws AudioException in case the file does not exist or cannot be opened
+     */
+    public void playFile(String fileName, String sink) throws AudioException {
+        File file = new File(System.getProperty(ConfigConstants.CONFIG_DIR_PROG_ARGUMENT) + File.separator + SOUND_DIR
+                + File.separator + fileName);
+        FileAudioStream is = new FileAudioStream(file);
+        play(is, sink);
+    }
+
+    /**
+     * Stream audio from the passed url using the default audio sink.
+     *
+     * @throws AudioException in case the url stream cannot be opened
+     */
+    public void stream(String url) throws AudioException {
+        stream(url, null);
+    }
+
+    /**
+     * Stream audio from the passed url to the given sink
+     *
+     * @param url The url to stream from
+     * @param sinkId The id of the audio sink to use or null
+     * @throws AudioException in case the url stream cannot be opened
+     */
+    public void stream(String url, String sinkId) throws AudioException {
+
+        AudioStream audioStream = new URLAudioStream(url);
+
+        AudioSink sink = null;
+        if (sinkId == null) {
+            sink = getSink();
+        } else {
+            sink = audioSinks.get(sinkId);
+        }
+
+        if (sink != null) {
+            try {
+                sink.process(audioStream);
+            } catch (UnsupportedAudioFormatException e) {
+                logger.error("Error playing '{}': {}", url, e.getMessage());
+            }
+        }
+    }
+
+    /**
+     * Retrieves the current volume of a sink
+     *
+     * @param sinkId the sink to get the volume for
+     * @return the volume as a value between 0 and 1
+     */
+    public float getVolume(String sinkId) {
+        AudioSink sink = null;
+        if (sinkId == null) {
+            sink = getSink();
+        } else {
+            sink = audioSinks.get(sinkId);
+        }
+
+        if (sink != null) {
+            try {
+                return sink.getVolume();
+            } catch (IOException e) {
+                logger.error("An exception occured while getting the volume of sink {} : '{}'", sink.getId(),
+                        e.getMessage());
+            }
+        }
+
+        return 0;
+
```

Comment:
As you already need to touch this file, I think this empty line could be removed.